### DISCUSSION TITLE
Reactjs website not indexable if deployed on Vercel

### DISCUSSION THREAD

**Discussion by hypetf** [vercel/community#351] [Help]
I have a custom domain and I've added it to my domains list on Vercel. It works, the URL is up and shows the content of my react website.

However it is not indexable on Google Search Console. The URL is not found by Google Search. I have a robots.txt but it isn't what it's blocking it neither any tags in the header.

When the latest commit is deployed on Vercel, it automatically creates a deployment URL and some other subdomains.

Those subdomains are not indexable because by default there's the header "X-Robots-Tag" set to "noindex". But if I add those subdomains (excluded the deployment URL which is not possible to) to the domains list, that subdomain becomes a production URL and with that URL now I can request indexing on Google Search Console. The URL is found by Google either way.

However I obviously don't want that vercel subdomain URL to be indexed but my custom one. What I've tried is setting that Production subdomain URL to redirect to my custom URL, however by doing that, that subdomain is no longer found by Google Search Console, because it is no longer a Production URL. And my custom domain remains (which has the flag of Production on the domains list), before and after, not found from Google Search Console.

So at this point I have no idea what I should do/try.

![Screenshot 2022-02-12 203920](https://user-images.githubusercontent.com/49875228/153727629-9004cc9c-e585-4ce6-9a77-a4eed285a399.png)
![Screenshot 2022-02-12 154418](https://user-images.githubusercontent.com/49875228/153727630-769424f5-f020-45d7-8af0-1af69396e74c.png)

**Comment by leerob** [comment]
Your custom domain should be an "alias" to your production deployment, which will have the correct indexing tags set. You are correct that preview deployments are not indexed on purpose. You'll want to ensure you're using a production deployment. 

More here: https://vercel.com/docs/concepts/deployments/automatic-urls

**Comment by hypetf** [comment]
It's a bit confusing. I already promoted my project to production, and it does display that it's a Production deployment. And under "domains", the custom one is shown. 
![image](https://user-images.githubusercontent.com/49875228/153729070-2aaf8583-918c-48e0-845d-5436a521807b.png)

And on the domains list, the custom domain is flagged as Production, same as the one autogenerated. The only difference is that the autogenerated URL is found by Google Search Console but the custom one isn't.
![image](https://user-images.githubusercontent.com/49875228/153729161-a6ee9690-9490-42b1-9412-aea7106d0e7e.png)

So I'm a bit confused about how would I get my custom domain be found by Google Search Console?

**Reply by leerob** [reply]
From the Vercel side, you are correctly configured then. You need to submit your custom domain for indexing in Google Search Console 👍

**Reply by hypetf** [reply]
That is the issue, the URL is not found. I was told it could take some time before it can be found, but 7 days passed since I bought the domain and deployed the website to it.
![image](https://user-images.githubusercontent.com/49875228/153729274-19947b1a-9af9-4f80-92d8-d4362defb234.png)

And if I try to Request indexing, I get this, which brings me back to the screenshot above.
![image](https://user-images.githubusercontent.com/49875228/153729286-7e31c464-903f-4a58-8812-d0527888f4d1.png)

**Comment by Valenmetal** [comment]
I have the same issue, my Production deployment isn't indexing. Did you found the solution?
